java/spark/src/test/java/com/lancedb/lance/spark/TestUtils.java

```diff
@@ -37,6 +37,12 @@ public static class TestTable1Config {
             Arrays.asList(1L, 2L, 3L, -1L),
             Arrays.asList(2L, 4L, 6L, -2L),
             Arrays.asList(3L, 6L, 9L, -3L));
+    public static final List<List<Long>> expectedValuesWithRowId =
+        Arrays.asList(
+            Arrays.asList(0L, 0L, 0L, 0L, 0L),
+            Arrays.asList(1L, 2L, 3L, -1L, 1L),
+            Arrays.asList(2L, 4L, 6L, -2L, 4294967296L),
+            Arrays.asList(3L, 6L, 9L, -3L, 4294967297L));
```

Comment:
I'm not sure if this works with Java's signed integers, but if possible I would represent the row ids in terms of their parts for clarity:
```suggestion
            Arrays.asList(2L, 4L, 6L, -2L, 1L << 32 + 0L),
            Arrays.asList(3L, 6L, 9L, -3L, 1L << 32 + 1L));
```

Reply:
It works, since I cast the `UInt8` into Long. If it's overflow, It will raise exception.